bool connected
bool mode
bool armed
